ansible_collections/arista/avd/plugins/plugin_utils/utils/avd_action_plugin/avd_action_plugin.py

```diff
@@ -0,0 +1,161 @@
+        if task_vars is None:
+            task_vars = {}
+
+        self.result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        # Prepare handlers, filters, and format based on logging config and task arguments
+        temp_handlers = []
+        if self._task.args.get("save_logs", False):
+            temp_handlers.append(SaveToResultHandler(result_dict=self.result))
+        if self._task.args.get("live_display", True):
+            temp_handlers.append(AnsibleDisplayHandler(display=Display()))
+
+        # Build the context data and log format string
+        context_data, format_parts = {}, []
+        if self._logging_config.add_role_context:
+            context_data["role_name"] = task_vars.get("ansible_role_name")
+            format_parts.append("[%(role_name)s] -")
+        if self._logging_config.add_hostname_context:
+            context_data["hostname"] = task_vars.get("inventory_hostname")
+            format_parts.append("<%(hostname)s>")
+
+        format_parts.append("%(message)s")
+        log_format = " ".join(format_parts)
+
+        temp_filters = [ContextFilter(context_data)] if context_data else []
+
+        try:
+            # Use the context manager to apply changes and ensure cleanup
+            with (
+                warnings.catch_warnings(record=True) as captured_warnings,
+                self._logging_context(temp_handlers=temp_handlers, temp_filters=temp_filters, log_format=log_format),
+            ):
+                # DeprecationWarning is ignored by default
+                warnings.simplefilter("always", DeprecationWarning)
+
+                # Run the plugin
+                self.main(task_vars)
+
+                # Process captured Python warnings and update the result object
+                if captured_warnings:
+                    self.result.setdefault("deprecations", [])
+                    self.result.setdefault("warnings", [])
+                    for w in captured_warnings:
+                        msg = str(w.message)
+                        if issubclass(w.category, DeprecationWarning):
+                            # AvdDeprecationWarning's are added from AvdSchemaTools with more context
+                            # This is a catch-all for other deprecations
+                            self.result["deprecations"].append({"msg": msg})
+                        else:
+                            # Catch-all for standard Python warnings from any library
+                            self.result["warnings"].append(msg)
+
+        except BaseException as exc:
+            # Recast errors as AnsibleActionFail
+            msg = f"Error during plugin execution: {exc}"
+            raise AnsibleActionFail(msg) from exc
+        else:
+            return self.result
+
+    @contextmanager
+    def _logging_context(
+        self,
+        temp_handlers: list[logging.Handler],
+        temp_filters: list[logging.Filter],
```

Comment:
pyright says you should do one of these


```suggestion
        temp_filters: list[ContextFilter],
```

or

```suggestion
        temp_filters: Sequence[Filter],
```

or type the list line 66 as

```
temp_filter: list[logging.Filter] = ...
```

Reply:
Fixed.